bindings/java/src/error.rs

```diff
@@ -15,26 +15,34 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use std::backtrace::{Backtrace, BacktraceStatus};
 use std::fmt::Debug;
 use std::fmt::Display;
 use std::fmt::Formatter;
 
 use jni::objects::JThrowable;
 use jni::objects::JValue;
 use jni::JNIEnv;
+
 use opendal::ErrorKind;
 
 pub(crate) struct Error {
     inner: opendal::Error,
+    backtrace: Backtrace,
 }
 
 impl Error {
-    pub(crate) fn unexpected(err: impl Into<anyhow::Error> + Display) -> Error {
+    pub(crate) fn new(inner: opendal::Error) -> Error {
         Error {
-            inner: opendal::Error::new(ErrorKind::Unexpected, &err.to_string()).set_source(err),
+            inner,
+            backtrace: Backtrace::capture(),
         }
     }
 
+    pub(crate) fn unexpected(err: impl Into<anyhow::Error> + Display) -> Error {
```

Comment:
Accept an `impl Into<anyhow::Error>` seems not cool. `opendal::Error` will capture the backtrace while creating. How about accpet a simple `&str` instead?

Reply:
Let's investigate this in another thread. I suggest it's a larger issue to dig and can cause a large amount of error handling code refactor.

Reply:
This is conveied from `pub fn set_source(mut self, src: impl Into<anyhow::Error>) -> Self {`. I want to set the source error.

Reply:
But it seems we also use the Display format for these sources only. I have an idea for improving this situation. Let's just do it in another thread.